### PR TITLE
Decompile reshow battle screen

### PR DESCRIPTION
**Pull request by DizzyEggg**
Changed labels:
sub_8031794 = BattleLoadOpponentMonSprite
sub_80318FC = BattleLoadPlayerMonSprite
sub_8031AF4 = LoadPlayerTrainerBankSprite
sub_8032350 = BattleLoadSubstituteSprite

Changed struct:
struct Struct2017800
{
    u8 invisible:1; // changed from unk0_0
    u8 unk0_1:1;
    u8 substituteSprite:1; // changed from unk0_2
    u8 unk0_3:1;
    u8 unk0_4:1;
    u16 transformedSpecies; // changed from unk2
};

### CODE REVIEW

File: src/reshow_battle_screen.c [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+static void CB2_ReshowBattleScreenAfterMenu(void);
+static bool8 LoadAppropiateBankSprite(u8 bank);
+void sub_807B184(u8 bank);
+void sub_807B508(u8 bank);
+void sub_807B06C(void);
+
+#define RESHOW_STATE 0x1FFFF
+#define HELPER_STATE 0x1FFFE
+
+struct BGCNT
```
Comment (camthesaxman) [inline src/reshow_battle_screen.c]: We already have macros for this in battle_anim.h.

File: src/reshow_battle_screen.c [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+void sub_8043DFC(u8 healthboxID);
+
+// this file's functions
+static void CB2_ReshowBattleScreenAfterMenu(void);
+static bool8 LoadAppropiateBankSprite(u8 bank);
+void sub_807B184(u8 bank);
+void sub_807B508(u8 bank);
+void sub_807B06C(void);
+
+#define RESHOW_STATE 0x1FFFF
```
Comment (camthesaxman) [inline src/reshow_battle_screen.c]: We usually do something like `#define gReshowState ewram[0x1FFFF]` to make it look like a normal variable and be more readable.

Comment (DizzyEggg) [inline src/reshow_battle_screen.c]: Is it something I HAVE to change? Personally I prefer the ewram version.

Comment (DizzyEggg) [inline src/reshow_battle_screen.c]: Alright, I changed it just like you said.

**Pull request merged**